### PR TITLE
[DOCS] Note the cluster settings API can override `elasticsearch.yml`

### PR DESCRIPTION
**Pull request by jrodewig**
### Preview
https://elasticsearch_61394.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/settings.html#cluster-setting-types

### CODE REVIEW

File: docs/reference/setup/configuration.asciidoc [hunk 1]
```diff
@@ -98,7 +98,8 @@ You can configure and update dynamic settings on a running cluster using the
 <<cluster-update-settings,cluster update settings API>>.
```
Comment (lcawl) [inline docs/reference/setup/configuration.asciidoc:98]: > You can configure and update dynamic settings...

It's covered in the API, but is it also worth highlighting here that dynamic settings can be either persistent (apply across restarts) or transient (don’t survive a full cluster restart)?  That's something you don't get with static settings.

File: docs/reference/setup/configuration.asciidoc [hunk 2]
```diff
@@ -98,7 +98,8 @@ You can configure and update dynamic settings on a running cluster using the
...
 +
 You can also configure dynamic settings locally on an unstarted or shut down
-node using `elasticsearch.yml`.
+node using `elasticsearch.yml`. However, any changes made using the cluster
+update settings API will override this file.
```
Comment (lcawl) [inline docs/reference/setup/configuration.asciidoc:102]: Not sure if it's more or less confusing, but here's another option:

```suggestion
node using `elasticsearch.yml`. However, the values shown in the <<cluster-get-settings,cluster get settings API>> 
override the values in the `elasticsearch.yml` configuration file.

**Comment by jrodewig** [comment]
Thanks for the feedback @lcawl. I've updated the PR to cover persistent and transient settings. I also simplified the order of precedence so its clear it applies for resets or setting configurations using multiple methods.

**Review by lcawl — APPROVED** [review]
So much clearer now, thanks!

**Review by pugnascotia — APPROVED** [review]

**Pull request merged**